ci/tasks.py

```diff
@@ -0,0 +1,259 @@
+    },
+)
+def diff(
+    c,
+    driver=None,
+    app=None,
+    outfile=None,
+):
+    """
+    Applies kubectl diff to manifests for a given channel.
+    """
+    assert c.calitp_config.git_root is not None
+    actual_driver = ReleaseDriver[driver] if driver else None
+    full_diff = ""
+    result: Result
+
+    for release in get_releases(c, driver=actual_driver, app=app):
+        if release.driver == ReleaseDriver.kustomize:
+            assert release.kustomize_dir is not None
+            kustomize_dir = c.calitp_config.git_root / Path(release.kustomize_dir)
+            result = c.run(f"kubectl diff -k {kustomize_dir}", warn=True)
+        elif release.driver == ReleaseDriver.helm:
+            assert release.helm_chart is not None
+            chart_path = c.calitp_config.git_root / Path(release.helm_chart)
+            c.run(f"helm dependency update {chart_path}")
+
+            with tempfile.TemporaryDirectory() as tmpdir:
+                manifest_path = Path(tmpdir) / Path("manifest.yaml")
+                kustomization_path = Path(tmpdir) / Path("kustomization.yaml")
+                values_str = " ".join(
+                    [
+                        f"--values {c.calitp_config.git_root / Path(values_file)}"
+                        for values_file in release.helm_values
+                    ]
+                )
+                # TODO: consider looking into https://github.com/databus23/helm-diff
+                c.run(
+                    f"helm template {release.helm_name} {chart_path} --namespace {release.namespace} {values_str} > {manifest_path}"
+                )
+                with open(kustomization_path, "w") as f:
+                    f.write(KUSTOMIZE_HELM_TEMPLATE.format(namespace=release.namespace))
+                result = c.run(f"kubectl diff -k {tmpdir}", warn=True)
+        else:
+            print(f"Encountered unknown driver: {release.driver}", flush=True)
+            raise RuntimeError
+
+        if result.exited != 0:
+            full_diff += result.stdout
+
+    msg = (
+        f"```{full_diff}```"
+        if full_diff
+        else f"No {driver if driver else 'manifest'} changes found for {c.calitp_config.channel}.\n"
+    )
+    if outfile:
+        print(f"writing {len(msg)=} to {outfile}", flush=True)
+        with open(outfile, "w") as f:
+            f.write(msg)
+    else:
+        print(msg, flush=True)
+
+
+# TODO: we may want to split up channels into separate files so channel is not an argument but a config file
+@task(parse_calitp_config, help=GENERIC_HELP)
+def release(
+    c,
+    driver=None,
+    app=None,
+):
+    """
+    Releases (i.e. deploys) apps into a specific channel.
+    """
+    assert c.calitp_config.git_root is not None
+    actual_driver = ReleaseDriver[driver] if driver else None
+
+    for release in get_releases(c, driver=actual_driver, app=app):
+        if release.driver == ReleaseDriver.kustomize:
+            assert release.kustomize_dir is not None
```

Comment:
just noting that this exact pattern (to build `kustomize_dir`) appears twice right? so yeah wondering if that can be simplified. and same for `chart_path`